geode-core/src/main/java/org/apache/geode/management/internal/configuration/mutators/ConfigurationManager.java

```diff
@@ -30,14 +32,16 @@
  * type {@link CacheElement}, which represents the configuration change.
  */
 @Experimental
-public interface ConfigurationManager<T extends CacheElement> {
+public interface ConfigurationManager<T extends RestfulEndpoint<R>, R extends RuntimeResponse> {
   void add(T config, CacheConfig existing);
 
   void update(T config, CacheConfig existing);
 
   void delete(T config, CacheConfig existing);
 
-  List<? extends T> list(T filterConfig, CacheConfig existing);
+  List<R> list(T filterConfig, CacheConfig existing);
 
-  T get(String id, CacheConfig existing);
+  R get(String id, CacheConfig existing);
```

Comment:
so the `get` method will return the runtime attributes as well. the original purpose of this method, is simply to get the static configuration from a group and check if they exists or not and check the configuration compatibility. I am not sure whether we want to change to this behavior.

Reply:
I added a `has` method for the purpose of checking existence, which is all the validators needed.  But I'm open to discussing whether we need separate `get` and `list` methods.  In my mind `get` was just a convenience for `list` /w id filter, I didn't realize it was intended for a totally different purpose.